modules/lsfilter/views/cmd/index.php

```diff
@@ -17,106 +17,219 @@
+		'method' => 'post'
+	) );
+	echo form::hidden( 'command', $command );
+	echo form::hidden( 'query', $set->get_query() );
+	echo "<table>";
+	foreach ( $command_info['params'] as $pname => $pdef ) {
+		$pdef = array_merge(array(
+			'name' => $pname,
+			'description' => '',
+			'option' => array(),
+			'default' => false
+		), $pdef);
+		echo "<tr>";
+		echo "<td>";
+		echo "<label for='field_$pname'>" . html::specialchars($pdef['name']) . "</label>";
+		echo "</td>";
+		echo "<td>";
+		switch ($pdef['type']) {
+			case 'string' :
+				echo form::input( array (
+					'class' => "input-wide autotest-required",
+					'name' => $pname,
+					'title' => _( 'Required field' ),
+					'id' => 'field_' . $pname
+				), (string) $pdef['default'] );
+				break;
+			case 'int' :
+			case 'float' :
+				echo form::input( array (
+					'class' => "input-wide autotest-required",
+					'name' => $pname,
+					'title' => _( 'Required field' ),
+					'id' => 'field_' . $pname
+				), (string) $pdef['default'] );
+				break;
+			case 'time' :
+				$default = "";
+				if ($pdef['default']) {
+					$default = date(date::date_format(), strtotime($pdef['default']));
+				}
+				echo form::input( array (
+					'class' => "input-wide autotest-date",
+					'name' => $pname,
+					'title' => _( 'Required field' ),
+					'id' => 'field_' . $pname
+				), $default );
+				break;
+			case 'duration' :
+				echo form::input( array (
+					'class' => "input-wide autotest-float",
+					'name' => $pname,
+					'title' => _( 'Required field' ),
+					'id' => 'field_' . $pname
+				), (string) $pdef['default'] );
+				break;
+			case 'select' :
+				echo form::dropdown( array (
+					'class' => "input-wide autotest-required",
+					'name' => $pname,
+					'title' => _( 'Required field' ),
+					'id' => 'field_' . $pname
+				), $pdef['option'] );
+				break;
+			case 'bool' :
+				echo form::checkbox(  array (
+					'name' => $pname,
+					'id' => 'field_' . $pname
+				), true, (bool) $pdef['default'], 'class="checkbox"' );
+				break;
+			case 'object':
+				$set = ObjectPool_Model::get_by_query($pdef['query']);
+				$objs = array(
+					'' => 'None'
+				);
+				foreach($set->it(array('key', 'readable_name'), array(), 1000) as $obj) {
+					/* @var $obj Object_Model */
+					$objs[$obj->get_key()] = $obj->get_readable_name();
+				}
+				echo form::dropdown(array(
+					'class' => "input-wide",
+					'name' => $pname,
+					'title' => _( 'Required field' ),
+					'id' => 'field_' . $pname,
+					'options' => $objs
+				));
+		}
+		if(!empty($pdef['description'])) {
+			echo "<br />";
+			echo html::specialchars($pdef['description']);
+		}
+		echo "</td>";
+		echo "</tr>";
 	}
-	if(!empty($pdef['description'])) {
-		echo "<br />";
-		echo html::specialchars($pdef['description']);
+	echo "</table>";
+	?>
+	<script>
+		$(document).ready(function(){
+			$("#command_form").submit();
+		});
+	</script>
+	<?php
+	echo form::close();
+} 
+// For other actions
+else {
```

Comment:
This `else` contains code that looks exactly like the code inside the `if` statement.

Is it possible to just print the form, as it's supposed to be (as in the `else` statement)? and delete the `if` statement.
In that case you can do an `if($command_info['name'] == 'Check now'){` just where the code differs. This makes it easier to see what's special for "check now", and we won't have to edit code in 2 places, if we want to do any changes.

Reply:
I thought about it but I thought it would be the same